network-store-client/src/main/java/com/powsybl/network/store/client/CollectionBuffer.java

```diff
@@ -81,4 +86,49 @@ void flush(UUID networkUuid, int variantNum) {
         updateResources.clear();
         removeResources.clear();
     }
+
+    private static <T extends IdentifiableAttributes> List<Resource<T>> cloneResources(Map<String, Resource<T>> resources, int newVariantNum,
+                                                        ObjectMapper objectMapper, Consumer<Resource<T>> resourcePostProcessor) {
+        // use json serialization to clone the resources of source collection
+        List<Resource<T>> clonedResources;
+        try {
+            var json = objectMapper.writeValueAsString(resources.values());
+            clonedResources = objectMapper.readValue(json, new TypeReference<>() {
+            });
+        } catch (JsonProcessingException e) {
+            throw new UncheckedIOException(e);
+        }
+        // reassign cloned resources to new variant number
+        for (Resource<T> clonedResource : clonedResources) {
+            clonedResource.setVariantNum(newVariantNum);
+            if (resourcePostProcessor != null) {
+                resourcePostProcessor.accept(clonedResource);
+            }
+        }
+        return clonedResources;
+    }
+
+    /**
+     * Buffer deep copy.
+     *
+     * @param objectMapper a object mapper to help cloning resources
+     * @param newVariantNum new variant num for all resources of the cloned buffer
+     * @param resourcePostProcessor a resource post processor
+     * @return the buffer clone
+     */
+    public CollectionBuffer<T> clone(ObjectMapper objectMapper, int newVariantNum, Consumer<Resource<T>> resourcePostProcessor) {
+        List<Resource<T>> clonedCreateResources = cloneResources(createResources, newVariantNum, objectMapper, resourcePostProcessor);
+        List<Resource<T>> clonedUpdateResources = cloneResources(updateResources, newVariantNum, objectMapper, resourcePostProcessor);
+
+        var clonedBuffer = new CollectionBuffer<>(createFct, updateFct, removeFct);
+        for (Resource<T> clonedResource : clonedCreateResources) {
+            clonedBuffer.createResources.put(clonedResource.getId(), clonedResource);
+        }
+        for (Resource<T> clonedResource : clonedUpdateResources) {
+            clonedBuffer.updateResources.put(clonedResource.getId(), clonedResource);
+        }
+        clonedBuffer.removeResources.addAll(removeResources);
```

Comment:
You should clone it too?

Reply:
```
    private final Set<String> removeResources = new HashSet<>();
```
It's just plain strings, we can just copy ?